WultraMobileTokenSDK/Operations/Model/UserOperation/TemplateParser/WMTTemplateListVisual.swift

```diff
@@ -0,0 +1,190 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions
+// and limitations under the License.
+//
+
+import Foundation
+
+/// `WMTTemplateListVisual` holds the visual data for displaying a user operation in a list view (table view/collection view)
+public struct WMTTemplateListVisual {
+    
+    /// The header of the cell
+    public let header: String?
+    
+    /// The title of the cell
+    public let title: String?
+    
+    /// The message (subtitle) of the cell
+    public let message: String?
+    
+    /// Predefined style of the cell on which the implementation can react
+    public let style: String?
+    
+    /// URL of the cell thumbnail
+    public let thumbnailImageURL: URL?
+    
+    /// Complete template from which the WMTTemplateListVisual was created
+    public let template: WMTTemplates.ListTemplate?
+    
+    public init(
+        header: String? = nil,
+        title: String? = nil,
+        message: String? = nil,
+        style: String? = nil,
+        thumbnailImageURL: URL? = nil,
+        template: WMTTemplates.ListTemplate? = nil
+    ) {
+        self.header = header
+        self.title = title
+        self.message = message
+        self.style = style
+        self.thumbnailImageURL = thumbnailImageURL
+        self.template = template
+    }
+}
+
+// MARK: WMTUserOperation List Visual preparation extension
+extension WMTUserOperation {
+
+    internal func prepareVisualListDetail() -> WMTTemplateListVisual {
+        let listTemplate = self.ui?.templates?.list
+        let attributes = self.formData.attributes
+        let headerAtrr = listTemplate?.header?.replacePlaceholders(from: attributes)
+        
+        var title: String? {
+            if let titleAttr = listTemplate?.title?.replacePlaceholders(from: attributes) {
+                return titleAttr
+            }
+            
+            if !self.formData.message.isEmpty {
+                return self.formData.title
+            }
+            
+            return nil
+        }
+
+        var message: String? {
+            if let messageAttr = listTemplate?.message?.replacePlaceholders(from: attributes) {
+                return messageAttr
+            }
+            
+            if !self.formData.message.isEmpty {
+                return self.formData.message
+            }
+            
+            return nil
+        }
+
+        var imageUrl: URL? {
+            if let imgAttr = listTemplate?.image,
+               let imgAttrCell = self.formData.attributes
+                                        .compactMap({ $0 as? WMTOperationAttributeImage })
+                                        .first(where: { $0.label.id == imgAttr }) {
+                return URL(string: imgAttrCell.thumbnailUrl)
+            }
+
+            if let imgAttrCell = self.formData.attributes
```

Comment:
Do we really want this - "If not defined, use the first image"? I would not assume anything here - what if the first image will be something not wanted?